### PR TITLE
Get data from dataplane inventory secret

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

**Review by cschwede — COMMENTED** [review]
With this patch applied, I see a panic in the operator:

```
2024-06-11T12:11:40.762Z	INFO	Observed a panic in reconciler: interface conversion: interface {} is []interface {}, not []string	{"controller": "swiftring", "controllerGroup": "swift.openstack.org", "controllerKind": "SwiftRing", "SwiftRing": {"name":"swift-ring","namespace":"openstack"}, "namespace": "openstack", "name": "swift-ring", "reconcileID": "778ce5b7-a06c-4d0d-ab35-53da8874af0d"}
panic: interface conversion: interface {} is []interface {}, not []string [recovered]
	panic: interface conversion: interface {} is []interface {}, not []string

goroutine 364 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.6/pkg/internal/controller/controller.go:116 +0x1fa
panic({0x1a72120, 0xc000442c00})
	/usr/lib/golang/src/runtime/panic.go:884 +0x213
github.com/openstack-k8s-operators/swift-operator/pkg/swiftring.DeviceList({0x1f6afc0, 0xc006ff06c0}, 0xc0006444d0, 0x8?)
	/remote-source/pkg/swiftring/funcs.go:110 +0xe6b
```

I'm looking into a possible fix.

**Comment by fao89** [comment]
The failure seems to be here:
```
nodeSetGroup.Vars[DataplaneDisks].([]byte)
```
where we try to cast the DataplaneDisks variable

File: pkg/swiftring/funcs.go [hunk 1]
```diff
@@ -85,46 +86,66 @@ func DeviceList(ctx context.Context, h *helper.Helper, instance *swiftv1beta1.Sw
...
-						return "", "", err
-					}
+	for _, secret := range secrets.Items {
+		inventory, err := ansible.UnmarshalYAML(secret.Data["inventory"])
+		if err != nil {
+			return "", "", err
+		}
+		nodeSetGroup := inventory.Groups[secret.Labels["openstackdataplanenodeset"]]
+
+		if slices.Contains(nodeSetGroup.Vars["edpm_services"].([]string), "swift") {
```
Comment (cschwede) [inline pkg/swiftring/funcs.go:110]: I had to change `[]string` to `[]interface {}` to make this work:

`if slices.Contains(nodeSetGroup.Vars["edpm_services"].([]interface {}), "swift") {
`

**Review by cschwede — CHANGES_REQUESTED** [review]
This worked in my tests with one minor change, please see my inline comment

**Review by cschwede — APPROVED** [review]
/lgtm

Tested, works fine for me - thx a lot for your help on this!

**Pull request merged**